### PR TITLE
Forget uavcan nodeid

### PR DESCRIPTION
**Pull request by rabad-matternet**
AVX-114 Align libcanard repository
Cleaning up loose ends so that we can work on the master branch.  In particular, committing Mitch's last commits to master.

@kuanli1 @afbowen @viet-nguyen-matternet
Currently, framework is linked to commit d1d7230.  bootloader is linked to an even older commit but on the same branch.  Commit d1d7230 is the same as branch forget-uavcan-nodeid.  This branch differs from master by only one commit (e6ded1b).  So I believe merging this to master and syncing framework to libcanard:master will have no affect on station_firmwares.

### CODE REVIEW

**Review by zalice — APPROVED** [review]

File: canard.h [hunk 1]
```diff
@@ -356,6 +356,12 @@ void canardSetLocalNodeID(CanardInstance* ins,
  */
 uint8_t canardGetLocalNodeID(const CanardInstance* ins);
 
+/**
+ * Forgets the current node ID value so that a new Node ID can be assigned.
+ * This is useful for Battery Packs that are never power cycled, but need to be added and removed from networks regularly.
+ */
+void canardForgetLocalNodeID(CanardInstance* ins);
```
Comment (viet-nguyen-matternet) [inline canard.h:363]: Why does `forget` have to be a keyword here? To me this implies setting it to NULL. Should this not simply be something more general such as:

```
 * Resets the current node ID value to initial broadcast id, so that a new Node ID can be assigned.
 * Example use case: battery Packs that are never power cycled, but need to be added and removed from networks regularly.

canardResetLocallNodeID(CanardInstance* ins) 
```

**Review by viet-nguyen-matternet — APPROVED** [review]

File: canard.h (hunk 1 above)
Comment (afbowen) [reply canard.h:363]: I also kind of like "reset" over "forget" because it does, in fact, reset to the default `node_id` value set in `canardInit()`, but I personally think the latter is acceptable since the function definition is so clear and a node can't _not_ have a node ID.

**Review by afbowen — APPROVED** [review]

File: canard.c [hunk 2]
```diff
@@ -131,6 +131,10 @@ uint8_t canardGetLocalNodeID(const CanardInstance* ins)
     return ins->node_id;
 }
 
+void canardForgetLocalNodeID(CanardInstance* ins) {
+    ins->node_id = CANARD_BROADCAST_NODE_ID;
```
Comment (kuanli1) [inline canard.c:135]: Just a quick concern I have about this specific branch.

For any CanardInstance all the following are being tracked:

https://github.com/matternet/libcanard/blob/d1d7230ab67048da60083465eda3cc818ddd1b90/canard.h#L258

```
struct CanardInstance
{
    uint8_t node_id;                                ///< Local node ID; may be zero if the node is anonymous

    CanardShouldAcceptTransfer should_accept;       ///< Function to decide whether the application wants this transfer
    CanardOnTransferReception on_reception;         ///< Function the library calls after RX transfer is complete

    CanardPoolAllocator allocator;                  ///< Pool allocator

    CanardRxState* rx_states;                       ///< RX transfer states
    CanardTxQueueItem* tx_queue;                    ///< TX frames awaiting transmission

    void* user_reference;                           ///< User pointer that can link this instance with other objects
};
```

I'm not at all familiar with canard.  Are we sure that this is all we need to do to is set node_id = CANARD_BROADCAST_NODE_ID?  

These changes are in a branch and after looking through all the call stacks this code isn't getting used by station firmwares AFAIK.  I personally do not know how well the changes that are on this branch have been tested and validated or if at all.  Would it be safer and cleaner not to include this branch (and the associated framework branch) into main?

File: canard.h (hunk 1 above)
Comment (rabad-matternet) [reply canard.h:363]: Perhaps.  We can certainly do that (and I'm not opposed to it).  This isn't my commit per se.  This was implemented by someone who is no longer here.  So I cannot answer "why?"

File: canard.c (hunk 2 above)
Comment (rabad-matternet) [reply canard.c:135]: @kuanli1 This function is being called in bmu_firmware, particularly when the BMU goes to sleep after being disconnected (from either aircraft or charger).  This call has been in place since Mitch committed it in March 2018 with commit e1a21aa. 
This change is in our BMU releases since v2.4.  I don't know what formal testing was done for this.  However, I have taken the current BMU firmware in and out of sleep and have not experienced any issues with neither going to sleep nor recovering from sleep and waking up.

In short, this change isn't needed in station_firmwares but it is needed in bmu_firmware.  Given the fact that all of our framework-based software (i.e. bmu_firmware and station_firmwares) already have this commit integrated (via libcanard branch forget-uavcan-nodeid), I think it's safe to merge into master and we all work with master from hereon.

Thoughts?

File: canard.h (hunk 1 above)
Comment (rabad-matternet) [reply canard.h:363]: @viet-nguyen-matternet @afbowen Do you guys want this name change (canardResetLocalNodeID) or is this okay in it's current state?

File: canard.c (hunk 2 above)
Comment (kuanli1) [reply canard.c:135]: I'm ok with merging this into the main branch with the information you provided.

**Review by kuanli1 — APPROVED** [review]

File: canard.h (hunk 1 above)
Comment (viet-nguyen-matternet) [reply canard.h:363]: I would prefer if it said: `canResetLocalNodeID`. If it isn't too much to change it. You may use your judgement to see if its worth it.
